src/js/services/conversation.js

```
@@ -113,84 +105,67 @@ export function handleConnectNotification(response) {
     };
 }
 
-export function sendChain(sendFn) {
+export function sendChain(sendFn, message) {
     return (dispatch, getState) => {
         const promise = dispatch(immediateUpdate(getState().user));
 
-        const enableScrollToBottom = (response) => {
-            dispatch(setShouldScrollToBottom(true));
-            return response;
-        };
-
-        if (getState().user.conversationStarted) {
-            return promise
+        const postSendHandler = (response) => {
+            return Promise.resolve(dispatch(onMessageSendSuccess(message, response)))
+                .then(() => dispatch(setShouldScrollToBottom(true)))
+                .then(() => dispatch(handleConnectNotification(response)))
                 .then(() => dispatch(connectFayeConversation()))
-                .then(() => sendFn())
-                .then(enableScrollToBottom)
-                .then((response) => dispatch(handleConnectNotification(response)));
-        }
+                .catch();
```

Please add a comment about why this swallows errors,